docs/extension-maintainers.md

```diff
@@ -0,0 +1,128 @@
+# PIE for Extension Maintainers
+
+Adding PIE support for your extension is relatively straightforward, and the
+flow is quite similar to adding a regular PHP package into Packagist.
+
+## Add a `composer.json` to your extension
+
+The first step to adding PIE support is adding a `composer.json` to your
+extension repository. Most of the typical fields are the same, with a few
+notable exceptions:
+
+ * The `type` MUST be either `php-ext` for a PHP Module (this will be most
+   extensions), or `php-ext-zend` for a Zend Extension.
+ * An additional `php-ext` section MAY exist.
+ * The Composer package name (i.e. the top level `name` field) for a PIE
+   extension cannot share the same Composer package name as a regular PHP
+   package, even if they have different `type` fields.
+
+### The `php-ext` definition
+
+#### `extension-name`
+
+The `extension-name` SHOULD be specified, and must conform to the usual extension
+name regular expression, which is defined in
+[\Php\Pie\ExtensionName::VALID_PACKAGE_NAME_REGEX](../src/ExtensionName.php).
+If the `extension-name` is not specified, the `extension-name` will be derived
+from the Composer package name, with the vendor prefix removed. For example,
+given a `composer.json` with:
+
+```json
+{
+    "name": "myvendor/myextension"
+}
+```
+
+The extension name would be derived as `myextension`. The `myvendor/` vendor
+prefix is removed.
+
+The `extension-name` MAY be prefixed with `ext-` as is a convention in Composer,
+but this is optional.
```

Comment:
I don't remember all of the context behind this decision, but as a best practice for new authors I think it's better to suggest people don't prefix their extension name, as `ext-` is only used to distinguish extensions from packages and libraries in an otherwise ambiguous name field. Since we're in the context of a PHP extension here, the `ext-` prefix is implied and should be omitted from the name.

Consider this:
```suggestion
The `extension-name` SHOULD NOT be prefixed with `ext-` as is a convention in Composer.
```

Instead of suggesting that users can use the prefix, this wording suggests that authors omit it, without saying that it's forbidden. I understand that leaves a grey area ("is it supported?"), but I'd much rather avoid a useless prefix.